src/Microsoft.AspNet.OData.Shared/Query/Expressions/FilterBinder.cs

```diff
@@ -600,7 +601,18 @@ private Expression ApplyNullPropagationForFilterBody(Expression body)
         /// <returns>The LINQ <see cref="Expression"/> created.</returns>
         public virtual Expression BindRangeVariable(RangeVariable rangeVariable)
         {
-            ParameterExpression parameter = _lambdaParameters[rangeVariable.Name];
+            ParameterExpression parameter = null;
+
+            // When we have a $this RangeVariable, we still create a $it parameter.
+            // i.e $it => $it instead of $this => $this
+            if (rangeVariable.Name == ODataThisParameterName)
```

Comment:
Currently, the FilterBinder is hardwired to create the parameterExpression using implicit $it. Since the implicit $it has same semantics as the $this, we don't need to put alot of effort in creating a $this parameterExpression.